account_reconciliation_widget/static/src/js/reconciliation/reconciliation_model.js

```diff
@@ -1603,13 +1603,10 @@ odoo.define("account.ReconciliationModel", function (require) {
                     );
                     if (matching && matching.length == 2) {
                         matching = matching[1].replace(
-                            new RegExp("\\D" + reconcileModel.decimal_separator, "g"),
-                            ""
-                        );
-                        matching = matching.replace(
                             reconcileModel.decimal_separator,
                             "."
                         );
+                        matching = matching.replace(new RegExp("[^0-9.-]", "g"), "");
```

Comment:
This is wrong. It breaks when "." is the thousands separator and "," decimal separator.
The code should be in the original local and the the regex should be just ```new RegExp("[^-\d" + reconcileModel.decimal_separator + "]", "g"),```

See #478 and the code in https://github.com/odoo/odoo/commit/d4da8be3669fbfdfdcad752af1f0311accaf34a3

Reply:
Maybe even without the `-`. I'm not entirely sure. All this looks like somewhat fragile code with a lot of overlap w.r.t. the host side python code.

Reply:
This module should be rewritten to use the python backend for most operations and js only for consuming it, probably on v16 migration as the odoo module was completly rewritten there.

Reply:
But yeah I did not see the problem for the thousand separator, indeed need to be changed, but i'm in favor of keeping the minus sign, as you can easily remove it using the main regex if you need to

Reply:
Ack

Reply:
We should fix it here for now on the best possible way, and don't wait on v16 refactoring, as all of that is enterprise and we can't access such code legally. The community part won't affect this JS part too much.